pybleau_plotting.puml

```diff
@@ -0,0 +1,144 @@
+@startuml
+
+'class DataElement
+'DataElement <|-- DataFramePlotManager
+'DataElement <|-- DataFrameAnalyzer
+class DataFrameAnalyzer {
+    DataFrame source_df
+    DataFrame filtered_df
+    DataFrame displayed_df
+    DataFrame summary_df
+    DataFrame summary_categorical_df
+    [DataFramePlotManager] plot_manager_list
+}
+DataFrameAnalyzer o--> DataFrame
+DataFrameAnalyzer "0..*" o--> DataFramePlotManager
+
+class DataFramePlotManager {
+    DataFrame data_source
+    [PlotDescriptor] contained_plots
+    DataFrameAnalyzer source_analyzer
+    IPlotTemplateInteractor template_interactor
+    ..Job Description..
+    * Manager to create, update, reorder, hide, delete plots.
+    * Provides a data model for a DataFrameAnalyzer.
+}
+DataFramePlotManager "0..*" o--> PlotDescriptor
+DataFramePlotManager o..> DataFrame
+DataFramePlotManager o--> DataFrameAnalyzer
+
+class BasePlotFactory {
+    ArrayPlotData plot_data
+    BaseXYPlotStyle plot_style
+    PlotContextMenuManager context_menu_manager
+    ..Job Description..
+    * Builds a chaco plot with generate_plot()
+    * Manipulates titles and ticks/labels for a plot.
+}
+BasePlotFactory o--> BaseXYPlotStyle
+BasePlotFactory o--> PlotContextMenuManager
+
+class StdXYPlotFactory {
+    OverlayPlotContainer plot
+    {BaseXYPlot} renderers
+    ..Job Description..
+    * Creates a dict of properties for a 2D plot
+    with one or more renderers and a legend.
+    * Initializes plot data from arrays.
+    * Adds tools (legend/context menu/zoom/pan) to a plot.
+    * Builds and adds/appends/removes/updates renderers
+}
+BasePlotContainer <|-- OverlayPlotContainer
```

Comment:
I shouldn't have used that `BasePlotContainer` class in the `PlotDescriptor` class, since we never us that, and we should simplify the graph by removing it and just mentioning `OverlayPlotContainer`. Will submit a PR to that effect...

Reply:
Thanks for making the change. I will go ahead and remove the BasePlotContainer from the UML.

Reply:
Sorry, I was wrong! There was a reason to make that the base class because some factories build and return an HPlotContainer that encapsulates an OverlayPlotContainer and its companion colorbar. I want to take my comment back...